openedx/tests/xblock_integration/test_review_xblock.py

```diff
@@ -0,0 +1,529 @@
+            'courseware_section',
+            kwargs={
+                'course_id': unicode(cls.course_review.id),
+                'chapter': 'Overview',
+                'section': 'Welcome',
+            }
+        )
+
+    def setUp(self):
+        super(TestReviewXBlock, self).setUp()
+
+        for idx, student in enumerate(self.STUDENTS):
+            username = 'u{}'.format(idx)
+            self.create_account(username, student['email'], student['password'])
+            self.activate_user(student['email'])
+
+        self.staff_user = GlobalStaffFactory()
+
+    def enroll_student(self, email, password, course):
+        """
+        Student login and enroll for the course
+        """
+        self.login(email, password)
+        self.enroll(course, verify=True)
+
+
+@attr(shard=1)
+@ddt.ddt
+class TestReviewFunctions(TestReviewXBlock):
+    """
+    Check that the essential functions of the Review xBlock work as expected.
+    Tests cover the basic process of receiving a hint, adding a new hint,
+    and rating/reporting hints.
+    """
+    def test_no_review_problems(self):
+        """
+        If a user has not seen any problems, they should
+        receive a response to go out and try more problems so they have
+        material to review.
+        """
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_actual)
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_review)
+
+        with self.store.bulk_operations(self.course_actual.id, emit_signals=False):
+            review_section_actual = ItemFactory.create(
+                parent=self.chapter_actual, display_name='Review Subsection'
+            )
+            review_unit_actual = ItemFactory.create(
+                parent=review_section_actual, display_name='Review Unit'
+            )
+
+            review_xblock_actual = ItemFactory.create(  # pylint: disable=unused-variable
+                parent=review_unit_actual,
+                category='review',
+                display_name='Review Tool'
+            )
+
+        # Loading the review section
+        response = self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': review_section_actual.location.name,
+            }
+        ))
+
+        expected_h2 = 'Nothing to review'
+        self.assertIn(expected_h2, response.content)
+
+    @ddt.data(5, 7)
+    def test_too_few_review_problems(self, num_desired):
+        """
+        If a user does not have enough problems to review, they should
+        receive a response to go out and try more problems so they have
+        material to review.
+
+        Testing loading 4 problems and asking for 5 and then loading every
+        problem and asking for more than that.
```

Comment:
Is this sentence still accurate?

Reply:
It is. When the client gets section1_actual, it loads 4 problems, but our number desired is 5 problems. Then when num_desired is > 6, we load all 6 problems in the course to show it is handled when you ask for more problems than possible.